pkg/userdata/ubuntu/userdata.go

```diff
@@ -353,29 +241,129 @@ write_files:
       sleep 1
     done
 
-- path: "/etc/default/kubelet-overwrite"
-  content: |
-    KUBELET_DNS_ARGS=
-    KUBELET_EXTRA_ARGS=--authentication-token-webhook=true \
-      {{- if .CloudProvider }}
-      --cloud-provider={{ .CloudProvider }} \
-      --cloud-config=/etc/kubernetes/cloud-config \
-      {{- end}}
-      --hostname-override={{ .MachineSpec.Name }} \
-      --read-only-port=0 \
-      --protect-kernel-defaults=true \
-      {{- if semverCompare "<1.11.0" .KubeletVersion }}
-      --resolv-conf=/run/systemd/resolve/resolv.conf \
-      {{- end }}
-      --cluster-dns={{ ipSliceToCommaSeparatedString .ClusterDNSIPs }} \
-      --cluster-domain=cluster.local
-{{ if semverCompare "<1.11.0" .KubeletVersion }}
-- path: "/etc/systemd/system/kubelet.service.d/20-extra.conf"
-  permissions: "0644"
+- path: "/etc/systemd/system/kubelet.service"
   content: |
+    [Unit]
+    Description=Kubernetes Kubelet
+    Documentation=https://github.com/kubernetes/kubernetes
+    After=containerd.service
+    Requires=containerd.service
+    
     [Service]
-    EnvironmentFile=/etc/default/kubelet
-{{ end }}
+    ExecStartPre=/sbin/modprobe br_netfilter
+    ExecStartPre=/bin/mkdir -p /etc/kubernetes/manifests
+    ExecStart=/usr/local/bin/kubelet \
```

Comment:
Uhm, are you sure this is a good idea? Maybe we can discuss this on monday but I'd prefer to use the offical unitfile[0] and maybe extend it, but not completely run our own stuff

[0] https://github.com/kubernetes/kubernetes/blob/master/build/debs/10-kubeadm.conf